components/api-controller/pkg/controller/v1alpha2/controller.go

```diff
@@ -256,6 +265,18 @@ func (c *Controller) validateAPI(newAPI *kymaApi.Api, apiStatusHelper *ApiStatus
 	return setStatus(kymaMeta.Successful)
 }
 
+func (c *Controller) validateVirtualService(newAPI *kymaApi.Api) error {
```

Comment:
I suggest something like
```
func (c *Controller) validateVirtualService(newAPI *kymaApi.Api) error {

	newApiServiceFullName := fmt.Sprintf("%s.%s", newAPI.Spec.Service.Name, newAPI.GetNamespace())
	
	for _, svc := range c.blacklistedServices {
		
		if svc == newApiServiceFullName {
			return fmt.Errorf("service %s has been blacklisted by BLACKLISTED_SERVICES env", newApiServiceFullName)
		}
	}
	return nil
}
``` 
This way we can avoid multiple conditions and parsing strings.